piff/input.py

```diff
@@ -176,422 +80,558 @@ def makeStars(self, logger=None):
+                    simage_file_name: '@input.image_file_name'
+
+           See the description of the GalSim config parser for more details about the various
+           types that are valid here.
+
+                `https://github.com/GalSim-developers/GalSim/wiki/Config-Values`_
+
+        There are many other optional parameters, which help govern how the input files are
+        read or interporeted:
+
+            :chipnum:       The id number of this chip used to reference this image [default:
+                            image_num]
+
+            :image_hdu:     The hdu to use in the image files. [default: None, which means use
+                            either 0 or 1 as typical given the compression sceme of the file]
+            :weight_hdu:    The hdu to use for weight images. [default: None, which means a weight
+                            image with all 1's will be automatically created]
+            :badpix_hdu:    The hdu to use for badpix images. Pixels with badpix != 0 will be given
+                            weight == 0. [default: None]
+            :noise:         Rather than a weight image, provide the noise variance in the image.
+                            (Useful for simulations where this is a known value.) [default: None]
+
+            :cat_hdu:       The hdu to use in the catalog files. [default: 1]
+            :x_col:         The name of the X column in the input catalogs. [default: 'x']
+            :y_col:         The name of the Y column in the input catalogs. [default: 'y']
+            :flag_col:      The name of a flag column in the input catalogs.  Anything with
+                            flag != 0 is removed from the catalogs. [default: None]
+            :use_col:       The name of a use column in the input catalogs.  Anything with
+                            use == 0 is removed from the catalogs. [default: None]
+            :sky_col:       The name of a column with sky values. [default: None]
+            :gain_col:      The name of a column with gain values. [default: None]
+            :sky:           The sky level to subtract from the image values. [default: None]
+                            Note: It is an error to specify both sky and sky_col. If both are None,
+                            no sky level will be subtracted off.
+            :gain:          The gain to use for adding Poisson noise to the weight map.  [default:
+                            None] It is an error for both gain and gain_col to be specified.
+                            If both are None, then no additional noise will be added to account
+                            for the Poisson noise from the galaxy flux.
+            :minsnr:        The minimum S/N ratio to use.  If an input star is too faint, it is
+                            removed from the input list of PSF stars.
+            :maxsnr:        The maximum S/N ratio to allow for any given star.  If an input star
```

Comment:
I did wrestle with this difference a bit, but I do think min and max are the right prefixes, since at the end, the stars do all fall into this range of snr.  We just do a different thing to get there at the high and low end.  We can't remove noise to lift low snr stars up to min_snr, so we have to just clip them.  But at the high end we can add noise, so no point in throwing out these stars.

Do you think there is any use case for people wanting to throw out high snr stars?  Maybe we could keep max_snr as the name, but add another option to specify whether high snr stars are clipped or have noise added.